src/Provider/AbstractProvider.php

```diff
@@ -98,7 +108,15 @@ public function __construct($options = [], HttpAdapterInterface $httpClient = nu
             }
         }
 
-        $this->setHttpClient($httpClient ?: new CurlHttpAdapter());
+        if (empty($collaborators['httpClient'])) {
```

Comment:
Good point. My point about the `foo` string is that there should probably be a check that it's a valid generator:

```php
if (!$collaborators['httpClient'] instanceof RandomFactory) {
    // throw new \InvalidArgumentException('Wha?');
}
```